### PR TITLE
Add queries fragments for product + image

### LINKED ISSUE #35: Add GraphQL Fragments
- [x] Move repeated GraphQL fields into GraphQL fragments, to make them reusable.

### PR DESCRIPTION
**Pull request by bilouStrike**
This PR aims to close #35

### CODE REVIEW

File: src/queries/get-product-slug.js [hunk 1]
```diff
@@ -1,16 +1,15 @@
 import gql from 'graphql-tag';
```
Comment (NicestRudeGuy) [inline src/queries/get-product-slug.js:1]: Now we can include gql from `@apollo/client` . can we have that in every query and remove gql-tag dep ?

File: src/queries/fragments/image.js [hunk 2]
```diff
@@ -0,0 +1,9 @@
+const ImageFragment = `
+    fragment ImageFragment on MediaItem {
+        uri
+        title
+        srcSet
+        sourceUrl
```
Comment (imranhsayed) [inline src/queries/fragments/image.js:6]: Can we also have altTxt?

**Review by imranhsayed — CHANGES_REQUESTED** [review]
Good work @bilouStrike 🥇 
Just a few updates

File: pages/product/[slug].js [hunk 3]
```diff
@@ -7,9 +7,8 @@ import {
 } from '../../src/queries';
 
 const Product = (props) => {
-    const {
-        data: { product }
-    } = props;
+
+    const { product } = props ? { ...props?.data } : null;
```
Comment (imranhsayed) [inline pages/product/[slug].js:11]: Please change this to :

```javascript

const Product = ({data}) => {
 const { product } = data || {}

```

**Review by imranhsayed — CHANGES_REQUESTED** [review]

File: src/queries/fragments/image.js (hunk 2 above)
Comment (bilouStrike) [reply src/queries/fragments/image.js:6]: Of course, it's provided as altText, i will update it

**Comment by imranhsayed** [comment]
@bilouStrike 
We can also do fragment for users https://github.com/rtCamp/wp-decoupled/blob/a529243dfaaec84d4950395fc0cf7a1aad03ca37/src/queries/auth/login.js#L10

**Comment by imranhsayed** [comment]
please resolve conflicts

**Review by NicestRudeGuy — APPROVED** [review]
**LGTM**

**Review by m-muhsin — APPROVED** [review]
Looks good!

**Pull request merged**